tests/emukit/multi_fidelity/test_non_linear_models.py

```diff
@@ -171,7 +171,7 @@ def wrap_func(x):
 
         def wrap_gradients(x):
             x_full = np.concatenate([x[None, :], [[2]]], axis=1)
-            return non_linear_model.get_prediction_gradients(x_full)[0]
+            return non_linear_model.get_prediction_gradients(x_full)[0].flatten()
```

Comment:
Are these (this one and below) calls to `flatten()` needed? We are actually trying to avoid things like flatten, ravel and atleast_2d to keep good track of dimensions. Otherwise these dimensionality bugs are pretty hard to spot. Calls to these functions are still ok, but only for a good reason. What is the reason here?

Reply:
that's a good point, we need to do that.

Reply:
It turns out the "flatten" calls are not now needed for the tests to pass, so I've reinstated the "main" version of these tests.